### PR TITLE
(CFACT-232) Move common CMake setup from cfacter to leatherman

### PR DESCRIPTION
**Pull request by branan**
Still left to do is the cpplint/cppcheck setup.

### CODE REVIEW

**Comment by branan** [comment]
Travis failure is because this repo isn't actually set up for coveralls yet

**Comment by MikaelSmith** [comment]
Added to coveralls. I can't seem to restart the Travis build however, you should re-run the debug build.

**Comment by branan** [comment]
@MikaelSmith ammended the latest commit just to bump the timestamp and change the SHA. That kicked travis

**Comment by branan** [comment]
@MikaelSmith I'd also like to try to get appveyor running as part of this PR, while we're here

**Comment by MikaelSmith** [comment]
Added an AppVeyor project too, just need an appveyor.yaml pushed up.

**Comment by branan** [comment]
whoops, left unity build stuff in appveyor (which doesn't exist here of course). Fixing...

**Comment by branan** [comment]
Looks like all the automated tests are working

air guitar

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+install:
+  - ps: Get-Date
+  - git submodule update --init --recursive
+  - ps: Get-Date
+
+  - ps: choco install 7zip.commandline -source https://www.myget.org/F/puppetlabs
+  - ps: choco install mingw-w64 -Version 4.8.3 -source https://www.myget.org/F/puppetlabs
+  - ps: $env:PATH = "C:\tools\ruby215\bin;C:\tools\mingw64\bin;" + $env:PATH
```
Comment (MikaelSmith) [inline appveyor.yml]: The ruby215 on PATH is extraneous.

**Comment by branan** [comment]
@MikaelSmith I just pushed up a small fixup commit. Don't want to rebase while you're commenting on individual SHAs. LMK when you're done and I'll rebase it into 2926147

**Comment by branan** [comment]
Ditto for the fixup_appveyor commit. Will squash when review is done

**Comment by MikaelSmith** [comment]
Mac
```
ld: unknown option: --whole-archive
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [tests/leatherman_test] Error 1
make[1]: *** [tests/CMakeFiles/leatherman_test.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by branan** [comment]
@MikaelSmith looks like it's a different linker flag on OSX: http://stackoverflow.com/a/10080549

I'll update

**Comment by MikaelSmith** [comment]
I found a similar link: http://stackoverflow.com/questions/16294842/how-to-disable-c-dead-code-stripping-in-xcode. Thanks.

**Comment by branan** [comment]
@MikaelSmith even better: Let's just do this in a less-stupid way

**Comment by MikaelSmith** [comment]
:+1: for not using archives.

**Comment by MikaelSmith** [comment]
That looks like it covers all my feedback.

**Comment by branan** [comment]
From a "cmake-purity" standpoint I don't exactly like passing files instead of targets from one subdir to another... but the archives were just too much of a hassle.

**Comment by branan** [comment]
@MikaelSmith fixup commits squahsed in. Should be good once the final commits finish running tests

**Comment by MikaelSmith** [comment]
Still works on Mac.

**Comment by branan** [comment]
And we're green!

**Pull request merged**